JASP-Engine/JASP/R/regressionlinearbayesian.R

```diff
@@ -611,1138 +384,1065 @@ RegressionLinearBayesian <- function (
+  posteriorSummaryTable$addColumnInfo(name="upperCri", title = "Upper", type="number", format="sf:4;dp:3", overtitle = overtitle)
+  
+  if (!status$ready) {
+    posteriorSummaryTable$setExpectedRows(1)
+    return()
+  } else if (status$error) {
+    posteriorSummaryTable$error <- "badData"
+    return()
+  }
+  
+  nRows <- length(unlist(options$covariates)) + 1 # 1 extra for intercept
+  posteriorSummaryTable$setExpectedRows(nRows)
+  
+  footnote <- posteriorSummary[["footnotes"]]
+  if (!is.null(footnote)) {
+    posteriorSummaryTable$addFootnote(footnote, symbol="<em>Warning.</em>")
+  }
+  
+  .basregFillTablePosteriorSummary(posteriorSummary, basregModel, jaspResults, options)
+  
+  return()
 }
 
-.plotDiagnostics.basReg <- function(bas_obj, status, perform, which) {
-	# Plot several different diagnostics of the BAS object
-	#
-	# Args:
-	#   bas_obj: bas object
-	#   status: current status of the analysis
-	#   perform: 'run' or 'init'
-	#		which: integer specifying which type of diagnostic to plot:
-	#					 1=residuals, 2=model probs, 3=model complexity, 4=inclusion probs
-	#
-	# Return:
-	#   list with plot data
-
-	titles <- c(
-		"Residuals vs Fitted",
-		"Model Probabilities",
-		"Log(P(data|M)) vs. model size",
-		"Inclusion Probabilities"
-	)
-
-	# if there were errors or we're in init, show just empty x and y-axes
-	emptyPlot <- .makeEmptyPlot.basReg(title = titles[which], status = status)
-
-	if (status$ready && perform == "run") {
-
-		plot <- list()
-		plot[["title"]] <- titles[which]
-
-		p <- try(silent = FALSE, expr = {
-
-			if (c(which) != 4) {
-				w <- 530
-				h <- 400
-			} else {
-				w <- 700
-				h <- 400
-			}
-
-			plotObj <- .plotBas.basReg(bas_obj, which = c(which))
-			content <- .writeImage(width = w, height = h, plot = plotObj)
-
-			plot[["convertible"]] <- TRUE
-			plot[["obj"]] <- content[["obj"]]
-			plot[["data"]] <- content[["png"]]
-			plot[["width"]] <- w
-			plot[["height"]] <- h
-			plot[["status"]] <- "complete"
-
-		})
-
-		if (isTryError(p)) {
-			errorMessage <- paste("Plotting not possible:", .extractErrorMessage(p))
-			plot <- emptyPlot
-			plot[["error"]] <- list(error="badData", errorMessage=errorMessage)
-		}
-
-	} else { # init or errors
-
-		plot <- emptyPlot
-
-	}
-
-	return(plot)
+.basregFillTablePosteriorSummary <- function(posteriorSummary, basregModel, jaspResults, options) {
+  
+  BFinclusion <- basregModel[["BFinclusion"]]
+  priorProbs <- basregModel[["priorprobsPredictor"]]
+  nModels <- basregModel[["n.models"]]
+  coef <- posteriorSummary[["coef"]]
+  coefficients <- posteriorSummary[["coefficients"]]
+  probne0 <- posteriorSummary[["probne0"]]
+  loopIdx <- posteriorSummary[["loopIdx"]]
+  confInt <- posteriorSummary[["conf95"]]
+  
+  topm <- order(-basregModel$postprobs)[1:nModels]
+  mostComplex <- which.max(lengths(basregModel$which)[topm])
+  
+  for (i in loopIdx) {
+    coefficient <- .clean(coefficients[i])
```

Comment:
same thing about `.clean`

Reply:
Sorry?

Reply:
It seems some comment disappeared. If I'm not mistaken, `.clean` is done automagically by jaspResults once you enter the data.

Reply:
also Joris just merged the error property PR so you could also try that out already

Reply:
I tested and .clean() is still necessary. For example `NULL` is passed on in json as `null` which javascript cannot deal with. I would also venture a guess that it also doesn't show a nice infinity symbol when `Inf` is passed.

I added the error handling. It seems to work nicely (nice job!), I have two  "issues" with it though:
1) `basReg$getError()` does not convey its meaning super nicely.  I guess this is kinda personal, something like`basReg$hasError()` or `basReg$error()` is clearer, but I understand why you named it this way -- to match the `setError()`.
2) we HAVE to use containers now to make it work, only the containers are always displayed as such in the output and this adds a visible grouping I don't think we really need. Also, and this is something you also encountered, you cannot use this container grouping to pass dependencies from the parent to the child, instead all children MUST have some sort of dependency set otherwise they get thrown out.